server/src/main/java/org/elasticsearch/index/mapper/vectors/DenseVectorFieldMapper.java

```diff
@@ -171,6 +172,17 @@ public Builder(String name, IndexVersion indexVersionCreated) {
                     }
                 }
             });
+            this.indexOptions.addValidator(v -> {
+                if ((v instanceof HnswIndexOptions hnswIndexOptions)
+                    && hnswIndexOptions.quantizationOptions != null
+                    && elementType.getValue() == ElementType.BYTE) {
+                    throw new IllegalArgumentException(
+                        "[element_type] cannot be [byte] when using quantization type ["
```

Comment:
Nitpick: Error message should state that float is the only one that is allowed (Uno reverse)

Reply:
Maybe? I am also thinking ahead for when we add half_float support, which will support int8 quantization.

Reply:
Fair.